rcl_action/include/rcl_action/goal_handle.h

```diff
@@ -211,6 +211,28 @@ RCL_WARN_UNUSED
 bool
 rcl_action_goal_handle_is_active(const rcl_action_goal_handle_t * goal_handle);
 
+/// Check if a goal can be transitioned to CANCELING in its current state.
+/**
+ * This is a non-blocking call.
+ *
+ * <hr>
+ * Attribute          | Adherence
+ * ------------------ | -------------
+ * Allocates Memory   | No
+ * Thread-Safe        | No
+ * Uses Atomics       | No
+ * Lock-Free          | Yes
+ *
+ * \param[in] goal_handle struct containing the goal and metadata
+ * \return `true` if the goal can be transitioned to CANCELING from its current state, or
+ * \return `false` otherwise, also
+ * \return `false` if the goal handle pointer is invalid
```

Comment:
@jacobperron nit: consider reversing the last two, so that otherwise is left on the last `\return` entry.

Reply:
Done 
3a9be73